### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by arabiu033**
## Overview

- Add authorization to your app using [CanCanCan](https://github.com/CanCanCommunity/cancancan).

## Workflow
Install CanCanCan in your project.
Add a role column to the users table. Remember to use a migration for this.
A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

**Comment by TedLivist** [comment]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @arabiu033 and @zakiyousufi,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a project that has failing linters

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by arabiu033** [comment]
This is embarrassing; I completely forgot I worked on the CSS file. @TedLivist :pray: I apologize.

File: app/models/user.rb [hunk 1]
```diff
@@ -15,6 +16,10 @@ def user_posts
     Post.where('author_id = ?', id).limit(3)
   end
 
+  def admin?
+    role == 'admin'
+  end
```
Comment (rloterh) [inline app/models/user.rb:21]: - Nice job for adding admin? function in user model! 👌

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -15,7 +15,6 @@ def new
   end
```
Comment (rloterh) [inline app/controllers/posts_controller.rb:18]: - [x] Kindly ensure to use _CanCanCan_ for Post authorization in _Post Controller_, which should be similar to:

```ruby
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Yousufi & Aminu, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  🏃 🏃‍♂️

## Highlights 👍  
- Good job so far implementing authorization rules!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

**You can also consider:**
- Kindly include details of the other co-contributor(s) in your ReadMe.

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (arabiu033) [reply app/controllers/posts_controller.rb:18]: @rloterh Alright let me check into that :+1:

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾🎉✔**
Hello @arabiu033 and @zakiyousufi 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Your linters are passing all the checks.
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**